source/as-promise/index.ts

```diff
@@ -172,7 +176,7 @@ export default function asPromise<T>(options: NormalizedOptions): CancelableRequ
 							retryOptions: options.retry,
 							error,
 							computedValue: 0
-						})
+						}) as number
```

Comment:
Why `as number`?

Reply:
Because it can be also `Promise<number>`

Reply:
But then `computedValue` is typed incorrectly, as it has to accept `number | Promise<number>`?

Reply:
I'll fix that.